### DISCUSSION TITLE
Create a react-hook-form error mapper

### DISCUSSION THREAD

**Discussion by nahtnam** [next-safe-action/next-safe-action#138] [Ideas]
A nice integration addition would be to convert `result.validationErrors`, `result.serverError`, `result.fetchError` into an object that we can pass to `react-hook-form`'s `useForm` function in the `errors` prop. That way the form would auto-populate with the server errors once the form is submitted

**Comment by TheEdoRan** [comment]
I think this a good suggestion. However, I'm not going to integrate third party libraries specific code in next-safe-action, because I don't feel it's the right place to do it. It's external to the library, it increases maintenance cost and would enable requests for any third party library out there. Instead, an helper/adapter package could be published on NPM and/or documentation updated. Feel free to publish it or open a PR to update docs, thanks!
